File: AI Kontrol Merkezi/Otomasyon & Scriptler/list_versions.swift
import Foundation

let path = "/Users/canerozyildirim/Library/Mobile Documents/iCloud~md~obsidian/Documents/caner/Ders Notları/Yetişkin Hastalıkları Uygulaması/02_Sunumlar/Konu1_Obezite_Sunum.html"
let url = URL(fileURLWithPath: path)

guard let versions = NSFileVersion.otherVersionsOfItem(at: url) else {
    print("No other versions found.")
    exit(0)
}

print("Found \(versions.count) versions:")
for (index, version) in versions.enumerated() {
    print("[\(index)] Date: \(String(describing: version.modificationDate))")
    let destPath = "/Users/canerozyildirim/.gemini/antigravity/brain/01100986-60f3-414a-aebe-02298ee852cd/scratch/Konu1_version_\(index).html"
    let destURL = URL(fileURLWithPath: destPath)
    do {
        if FileManager.default.fileExists(atPath: destPath) {
            try FileManager.default.removeItem(at: destURL)
        }
        try FileManager.default.copyItem(at: version.url, to: destURL)
        print(" -> Copied to \(destPath)")
    } catch {
        print(" -> Error copying version \(index): \(error)")
    }
}

File: AI Kontrol Merkezi/Otomasyon & Scriptler/list_versions.py
import objc
from Foundation import NSFileVersion, NSURL

path = "/Users/canerozyildirim/Library/Mobile Documents/iCloud~md~obsidian/Documents/caner/Dersler/Yetişkinlerde Beslenme Tedavisi Uygulaması/Sunumlar/Konu1_Obezite_Sunum.html"
url = NSURL.fileURLWithPath_(path)

versions = NSFileVersion.otherVersionsOfItemAtURL_(url)
if not versions:
    print("No other versions found.")
else:
    for i, v in enumerate(versions):
        print(f"[{i}] {v.modificationDate()}")
        dest_url = NSURL.fileURLWithPath_(f"/Users/canerozyildirim/.gemini/antigravity/brain/01100986-60f3-414a-aebe-02298ee852cd/scratch/Konu1_version_{i}.html")
        from Foundation import NSFileManager
        fm = NSFileManager.defaultManager()
        fm.removeItemAtURL_error_(dest_url, None)
        success, err = fm.copyItemAtURL_toURL_error_(v.URL(), dest_url, None)
        if success:
            print(f" -> Copied version {i}")
        else:
            print(f" -> Failed to copy version {i}: {err}")
